### ISSUE TITLE
TypeError: argument of type 'NoneType' is not iterable during docker image update

### ISSUE THREAD

**Issue by criemen** [ansible/ansible-modules-core#4802]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

docker module
##### ANSIBLE VERSION

```
root@c51cf36122e4:/# ansible --version
ansible 2.0.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Debian Linux (both)
##### SUMMARY

When updating a running docker container with a new image (which was pushed to the registry), since today we have encountered these errors:
They only go away when manually stopping the container and removing the image.
This happens on all 5 servers to which the same image is deployed.
Unfortunately, I do not know if anything has changed in the setup, because at face it doesn't look like it, but a week ago the same ansible job worked.

fatal: [somehostname.local]: FAILED! => {"changed": false, "failed": true, "module_stderr": "", "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1473707610.26-10297167272284/docker\", line 3701, in <module>\r\n    main()\r\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1473707610.26-10297167272284/docker\", line 1738, in main\r\n    reloaded(manager, containers, count, name)\r\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1473707610.26-10297167272284/docker\", line 1595, in reloaded\r\n    containers.refresh()\r\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1473707610.26-10297167272284/docker\", line 1529, in refresh\r\n    self.deployed = self.manager.get_deployed_containers()\r\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1473707610.26-10297167272284/docker\", line 1345, in get_deployed_containers\r\n    repo_tags = self.get_image_repo_tags()\r\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1473707610.26-10297167272284/docker\", line 1008, in get_image_repo_tags\r\n    if resource in image.get('RepoTags', []):\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "parsed": false}
##### STEPS TO REPRODUCE

```

```

Since our setup is pretty involved and relies (besides other things) on our local docker registry, I can't really post an easy example playbook.
If this issue can't be fixed with the tracebooks provided or easy printf patches to the ansible source, I can invest time in reducing this, but this would be a non-trivial thing to do.

##### ACTUAL RESULTS

```

```

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by kravietz** [comment]
Same here with:

```
ansible==2.1.1.0
docker-py=1.10.3
```

Sample playbook rule that triggers this error:

```
 docker: name="{{item.name}}" image="{{item.image}}" state=stopped
  with_items:
  - { "name": "nginx", "image": "kravietz/webcookies:nginx" }
```

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by zfil** [comment]
wontfix
this module is deprecated
@cove, @softzilla, @zfil, @ThomasSteinbach guys, in general I suggest we stop docker modules needing docker-py to rely only on native docker component (like compose)

**Comment by criemen** [comment]
And which module exactly should we use instead?
Our whole deployment is based on docker, ansible and this module...

**Comment by zfil** [comment]
Use all the new docker_\* modules.

Personally I like docker_service as is takes docker compose file et don't require docker-py to be deployed

**Comment by seenickcode** [comment]
@zfil if the 'docker' module is deprecated, could you please add a clear/er warning on the documentation's page?  I think this would save devs hours of work trying to hack and get this to work properly with docker-py

**Comment by Akasurde** [comment]
@seenickcode Documentation does tell about `docker` module deprecation - [here](https://docs.ansible.com/ansible/devel/modules/docker_module.html#deprecated)

**Comment by seenickcode** [comment]
Right. I know that but it was not at all obvious that's what I was trying
to say. Would be great if the site can be more explicit about
troublesome/better options for things that are deprecated.

On Fri, Feb 16, 2018 at 10:05 AM, Abhijeet Kasurde <notifications@github.com

**Comment by webknjaz** [comment]
@seenickcode this is on the very top of the page:
```
DEPRECATED
Removed in Ansible:  | version: 2.4
Why: | Replaced by dedicated modules.
Alternative: | Use module_docs/docker_container and module_docs/docker_image instead.
```
Looks clear to me. What would you change to make it better? Any ideas?
